pkg/runtime/framework/plugins/mpi/mpi.go

```diff
@@ -84,15 +84,21 @@ func (m *MPI) Validate(_ client.Object, runtimeInfo *runtime.Info, _, newJobObj
 	if runtimeInfo == nil || runtimeInfo.RuntimePolicy.MLPolicySource == nil || runtimeInfo.RuntimePolicy.MLPolicySource.MPI == nil {
 		return nil, allErrs
 	}
-
 	specPath := field.NewPath("spec")
-	if newJobObj.Spec.Trainer != nil && newJobObj.Spec.Trainer.NumProcPerNode != nil {
-		numProcPerNodePath := specPath.Child("trainer").Child("numProcPerNode")
+	if newJobObj.Spec.Trainer.NumProcPerNode != nil {
```

Comment:
```suggestion
	if newJobObj.Spec.Trainer != nil && newJobObj.Spec.Trainer.NumProcPerNode != nil {
```

We still need this check.